aptos-move/aptos-gas-meter/src/traits.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_gas_algebra::{Fee, FeePerGasUnit, Gas, GasExpression, GasScalingFactor, Octa};
+use aptos_gas_schedule::VMGasParameters;
+use aptos_types::{
+    contract_event::ContractEvent, state_store::state_key::StateKey, write_set::WriteOp,
+};
+use aptos_vm_types::storage::StorageGasParameters;
+use move_binary_format::errors::{Location, PartialVMResult, VMResult};
+use move_core_types::gas_algebra::{InternalGas, InternalGasUnit, NumBytes};
+use move_vm_types::gas::GasMeter as MoveGasMeter;
+use std::fmt::Debug;
+
+/// An abstraction of the part of the gas meter that manages the core algebra,
+/// defining how the gas charges should be handled.
+pub trait GasAlgebra {
```

Comment:
I don't understand the structure. If it's an algebra (as we discussed before I consider "algebras" over engineering but that's not the point here) then I dont understand what feature_version/charge_execution etc are doing in this trait. That's not algebra